osu.Game/Beatmaps/WorkingBeatmap.cs

```diff
@@ -28,8 +28,10 @@ public abstract class WorkingBeatmap : IWorkingBeatmap
     {
         public readonly BeatmapInfo BeatmapInfo;
 
+        // ReSharper disable once FieldHidesInterfacePropertyWithDefaultImplementation
         public readonly BeatmapSetInfo BeatmapSetInfo;
 
+        // ReSharper disable once FieldHidesInterfacePropertyWithDefaultImplementation
         public readonly BeatmapMetadata Metadata;
```

Comment:
Not sure about this. We can remove the default implementation if preferred.

Reply:
Can you use `new` instead of the disable?

Reply:
Unfortunately doesn't work for interface methods.

Reply:
How do you feel about just implementing explicitly:
```diff
diff --git a/osu.Game/Beatmaps/WorkingBeatmap.cs b/osu.Game/Beatmaps/WorkingBeatmap.cs
index 51eea94d3a..682eaded1e 100644
--- a/osu.Game/Beatmaps/WorkingBeatmap.cs
+++ b/osu.Game/Beatmaps/WorkingBeatmap.cs
@@ -27,11 +27,7 @@ namespace osu.Game.Beatmaps
     public abstract class WorkingBeatmap : IWorkingBeatmap
     {
         public readonly BeatmapInfo BeatmapInfo;
-
-        // ReSharper disable once FieldHidesInterfacePropertyWithDefaultImplementation
         public readonly BeatmapSetInfo BeatmapSetInfo;
-
-        // ReSharper disable once FieldHidesInterfacePropertyWithDefaultImplementation
         public readonly BeatmapMetadata Metadata;
 
         protected AudioManager AudioManager { get; }
@@ -232,6 +228,10 @@ private Task<IBeatmap> loadBeatmapAsync()
 
         IBeatmapInfo IWorkingBeatmap.BeatmapInfo => BeatmapInfo;
 
+        IBeatmapSetInfo IWorkingBeatmap.BeatmapSetInfo => BeatmapSetInfo;
+
+        IBeatmapMetadataInfo IWorkingBeatmap.Metadata => Metadata;
+
         public IBeatmap Beatmap
         {
             get
```
Otherwise, I think there may be a nullability issue here - doesn't `((IWorkingBeatmap)beatmap).Metadata` violate the non-nullable constraint of `IBeatmapInfo.Metadata` given the `WorkingBeatmap` ctor:
```
BeatmapInfo = beatmapInfo;
Metadata = beatmapInfo.Metadata ?? BeatmapSetInfo?.Metadata ?? new BeatmapMetadata();
```

Reply:
Yeah, I'm fine with doing that. Will make the changes.